src/term/mod.rs

```diff
@@ -908,10 +907,23 @@ impl Term {
                 if cols.start >= line_end {
                     None
                 } else {
-                    for cell in &grid_line[cols.start..line_end] {
+                    let mut tab_mode = false;
+
+                    for col in IndexRange::from(cols.start..line_end) {
+                        let cell = grid_line[col];
+                        if tab_mode {
+                            if tabs[col] {
+                                tab_mode = false;
+                            } else if cell.c == ' ' {
+                                continue;
+                            }
+                        }
```

Comment:
It would probably be good to document this block because it's not obvious that it skips whitespaces until the next tabstop.

Reply:
@chrisduerr If you mean the way a tab can be partially selected as if it were a bunch of spaces, I totally agree. I've been working on a fix that will have to be a separate PR.

Reply:
Yeah that's what I'm talking about.